gr-qtgui/lib/TimeRasterDisplayPlot.cc

```diff
@@ -38,15 +38,38 @@ namespace pt = boost::posix_time;
  **********************************************************************/
 class QwtXScaleDraw : public QwtScaleDraw, public TimeScaleData
 {
+protected:
+    double d_start_value;
+    double d_end_value;
+    double d_delta_value;
+    int d_max_scale;
+
 public:
-    QwtXScaleDraw() : QwtScaleDraw(), TimeScaleData() {}
+    QwtXScaleDraw(double start_value = 0.0, double end_value = 0.0, int max_scale = 1024)
+        : QwtScaleDraw(),
+          TimeScaleData(),
+          d_start_value(start_value),
+          d_end_value(end_value),
+          d_max_scale(max_scale)
+    {
+        d_delta_value = d_end_value - d_start_value;
```

Comment:
Autoscales on any start_value == end_value. Not wrong, but I can picture the user who puts 10.0 in for both and has no clue what happened.